src/main/java/com/yelp/nrtsearch/server/luceneserver/field/BooleanFieldDef.java

```diff
@@ -140,25 +140,19 @@ public String getType() {
   }
 
   @Override
-  public Query getTermQuery(TermQuery termQuery) {
-    if (!isSearchable()) {
-      throw new IllegalStateException(
-          "Field " + getName() + " is not searchable, which is required for TermQuery");
-    }
+  public TermQuery.TermTypesCase getTermQueryType() {
+    return TermQuery.TermTypesCase.BOOLEANVALUE;
+  }
 
-    boolean termValue;
-    switch (termQuery.getTermTypesCase()) {
-      case TEXTVALUE:
-        termValue = parseBooleanOrThrow(termQuery.getTextValue());
-        break;
-      case BOOLEANVALUE:
-        termValue = termQuery.getBooleanValue();
-        break;
-      default:
-        throw new IllegalArgumentException(
-            "BOOLEAN field does not support term type: " + termQuery.getTermTypesCase());
-    }
-    String indexTermValue = termValue ? "1" : "0";
```

Comment:
FYI - BREAKING CHANGE
Removing support for TEXTVALUE in BooleanFieldDef means existing use cases will throw errors when inputting strings `"false"` or `"true"`.